shepherd_bms/include/datastructs.h

```diff
@@ -122,4 +122,83 @@ struct AccumulatorData_t
 
 };
 
+/**
+ * @brief Represents the state of the BMS fault timers
+ */
+typedef enum
+{
+	BEFORE_TIMER_START,
+	DURING_EVAL
+
+} TSTimerEvalState;
+
+/**
+ * @brief Represents a timer that can be in one of three states
+ */
+struct tristate_timer : public Timer
```

Comment:
You can inherit from a class with a struct ????

Reply:
yeah turns out c++ backend considers them the exact same thing with the ony differnce being the default member privacy, but they are fully compatible